### ISSUE TITLE
Buffer is not changing after switch in branch with similar names

### ISSUE THREAD

**Issue by joacohoyos** [ThePrimeagen/git-worktree.nvim#41]
Hi, today i was trying to switch between to branches with similar names. In my specific case they were **cancel-mail-fix** and **cancel-mail-fix-refactor**. 
Basically I could go from cancel-mail-fix to cancel-mail-fix-refactor and my buffer will update but not the other way.

I tried to reproduce the error in new bare repo and this were my results. The branches were named as follow:
- master
- a
- a-d

**Results**
- master to a **OK**
- master to a-d **OK**
- a to master **OK**
- a to a-d **OK**
- a-d to master **OK**
- a-d to a **ERROR**

Basically as you can see i went back and forth between the 3 branches and the buffer was updating correctly in all cases except when i was trying to go from a-d to a. In that case I got stuck in branch a-d.

I look at the logs but they were empty, if someone can tell me what to do in other to log everything I am happy to try again and provide the logs

**Comment by polarmutex** [comment]
by default logging is set to warn

call the following to enable logging temporarily

GIT_WORKTREE_NVIM_LOG=trace nvim

log file will be located at the following folder `":echo stdpath("cache")"

I suspect the issue is when we determine if the git worktree exits. I have not had time to test this so far today

thanks for reporting this
